fix: prepare jsHandle.uploadFile for CDP Page.handleFileChooser removal

mathiasbynens opened the pull request:
https://chromium-review.googlesource.com/c/chromium/src/+/1935410 removes `Page.handleFileChooser` from the CDP. This patch prepares `jsHandle.uploadFile` for that change while preserving existing semantics as much as possible.

mathiasbynens committed:
fix: prepare jsHandle.uploadFile for CDP Page.handleFileChooser removal

https://chromium-review.googlesource.com/c/chromium/src/+/1935410
removes Page.handleFileChooser from the CDP.

pavelfeldman commented on lib/JSHandle.js:
```diff
@@ -312,9 +315,21 @@ class ElementHandle extends JSHandle {
-    await this._client.send('DOM.setFileInputFiles', { objectId, files });
+    const promises = filePaths.map(filePath => readFileAsync(filePath, 'utf8'));
+    const buffers = await Promise.all(promises);
+    const fileContents = buffers.map(buffer => buffer.toString());
```
`buffer.toString('base64')` to support binary data.

pavelfeldman commented on lib/JSHandle.js:
```diff
@@ -312,9 +315,21 @@ class ElementHandle extends JSHandle {
+      for (let i = 0; i < filePaths.length; i++) {
+        const name = filePaths[i];
+        const content = fileContents[i];
+        const file = new File([content], name);
```
```
const result = await fetch(`data:${files[i].mimeType};base64,${files[i].content}`);
const file = new File([await result.blob()], files[i].name);
```

mathiasbynens replied:
Where are you getting `mimeType` from? Are you suggesting we introduce a dependency on `mime-types`, or am I missing some trick?

mathiasbynens committed:
fix: improve binary file support

UTF-8-decoding the input file could fail for binary files, and so we
now read the raw file buffer and base64-encode it. To base64-decode it
within the page context, we use the Fetch API in combination with a
data URL. This requires knowing the proper MIME type for the input
file, which we now figure out using the new mime-types dependency.

hanselfmu reviewed (commented):
LGTM

hanselfmu reviewed (approved):
LGTM

JoelEinbinder commented on lib/JSHandle.js:
```diff
@@ -312,9 +311,35 @@ class ElementHandle extends JSHandle {
+        dt.items.add(file);
+      }
+      element.files = dt.files;
+      element.dispatchEvent(new Event('input', { bubbles: true }));
```
Is there any way we can avoid the fake event here?
